addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5250,51 +5279,88 @@ registry.layout_column = SnippetOptionWidget.extend({
+            for (let i = 0; i < itemsDelta; i++) {
                 await new Promise(resolve => {
-                    this.trigger_up('clone_snippet', {$snippet: $lastColumn, onSuccess: resolve});
+                    this.trigger_up('clone_snippet', {$snippet: $(lastEl), onSuccess: resolve});
                 });
             }
-        } else {
-            var self = this;
-            for (const el of $row.children().slice(count)) {
+        } else if (itemsDelta < 0 && prevNbRows < 2) {
+            const self = this;
+            for (const el of [...rowEl.children].slice(itemsDelta)) {
                 await new Promise(resolve => {
                     self.trigger_up('remove_snippet', {$snippet: $(el), onSuccess: resolve, shouldRecordUndo: false});
                 });
             }
         }
 
-        this._resizeColumns($row.children());
+        this._resizeColumns(rowEl.children, nbColumns);
         this.trigger_up('cover_update');
     },
     /**
-     * Resizes the columns so that they are kept on one row.
+     * Resizes the columns according to mobile or desktop view.
      *
      * @private
-     * @param {jQuery} $columns - the columns to resize
-     */
-    _resizeColumns: function ($columns) {
-        const colsLength = $columns.length;
-        var colSize = Math.floor(12 / colsLength) || 1;
-        var colOffset = Math.floor((12 - colSize * colsLength) / 2);
-        var colClass = 'col-lg-' + colSize;
-        _.each($columns, function (column) {
-            var $column = $(column);
-            $column.attr('class', $column.attr('class').replace(/\b(col|offset)-lg(-\d+)?\b/g, ''));
-            $column.addClass(colClass);
+     * @param {HTMLCollection} columnEls - the elements to resize
+     * @param {integer} nbColumns - the number of wanted columns
+     */
+    _resizeColumns(columnEls, nbColumns) {
+        const isMobile = weUtils.isMobile(this);
+        const sameNbOfColsMobileDesktop = nbColumns === this._getNbColumns(!isMobile);
+        const resolutionModifier = isMobile ? '' : '-lg';
+        const itemSize = Math.floor(12 / nbColumns) || 1;
+        const firstItemOffset = this._getItemOffset(itemSize, nbColumns);
+        const resolutionRegex =
+            new RegExp(`\\b(col|offset)(${resolutionModifier})?(-\\d{1,2})?(?![\\w-])`,'g');
+
+        // When the desktop view changes (adds or removes items), the mobile
+        // view is reset accordingly.
+        let resetMobileItemSize;
+        if (!isMobile && !sameNbOfColsMobileDesktop) {
+            const mobileItemSize = this._getItemSize(columnEls[0]).mobile;
+            // Gets the closest denominator of 12.
+            resetMobileItemSize = 12 / (Math.round(12 / mobileItemSize) || 1);
+        }
+
+        [...columnEls].forEach(columnEl => {
+            columnEl.className = columnEl.className
+                .replace(resolutionRegex, '')
+                .replace(/\boffset-lg-0\b/, '');
```

Comment:
Is that just to remove the `offset-lg-0` class ? If yes you can just do `columnEl.classList.remove('offset-lg-0')`

Reply:
That's when you reorganize the code several times and in the end you don't see the obvious. 🤦